### DISCUSSION TITLE
How to add static Identifier fields in query columns

### DISCUSSION THREAD

**Discussion by meneer-code** [grafana/grafana-infinity-datasource#231] [Q&A]
Hi there

I'm making multiple queries to the same URL (just a different path variable in the url), but the external API does not provide an identifier in the response, so I'm getting this back:

The API requests differ only in the URL, everything else is the same:

Query 1:
https://api.opensea.io/api/v1/collection/animo-official/stats
```
{
    "stats": {
        "one_day_volume": 117,
        "one_day_change": 0.86,
        "one_day_sales": 493.0,
    }
}
```

Query 2:
https://api.opensea.io/api/v1/collection/morethangamersnftmtg/stats
```
{
    "stats": {
        "one_day_volume": 100,
        "one_day_change": 0.6,
        "one_day_sales": 49,
    }
}
```

Query 3:
https://api.opensea.io/api/v1/collection/alienfrensnft/stats
```
{
    "stats": {
        "one_day_volume": 50,
        "one_day_change": 0.1,
        "one_day_sales": 4,
    }
}
```

After merging the columns successfully, I get this:
![image](https://user-images.githubusercontent.com/15004267/148344322-35107c0d-53b5-4620-b7bd-bf4e4734e205.png)

The problem is that there is no way to define a static column with the query name or URL. I would ideally like to add a "Title" column stating Query 1, Query 2, Query 3

Is this possible, or should I set up my process differently?

**Comment by yesoreyeram** [comment]
I don't think it is possible to setup a static field. May be a feature request..

**Comment by iiojj2** [comment]
would like this feature too... i.e. diff by query label, or by URI itself or be able to add static field to a response on Grafana side
Accessing URI will help building dataLinks

**Comment by segevfiner** [comment]
Can just be done using JSONata, e.g. `$merge([$, { "foo": "bar" }])` but I'm not sure there are any builtin variables with stuff like the request URL and other details to avoid duplication.

**Reply by segevfiner** [reply]
And actually just computed columns, with a static value expression. e.g. `"bla"`. I think the syntax is dependant on the selected parser.
